non-critical-infra/modules/matrix-synapse.nix

```diff
@@ -0,0 +1,112 @@
+{ config
+, pkgs
+, ...
+}:
+
+{
+  imports = [
+    ./nginx.nix
+    ./postgresql.nix
+  ];
+
+  fileSystems."/var/lib/matrix-synapse" = {
+    device = "zroot/root/matrix-synapse";
+    fsType = "zfs";
+    options = [
+      "zfsutil"
+    ];
+  };
+
+  services.postgresql = {
+    ensureUsers = [ {
+      name = "matrix-synapse";
+      ensureDBOwnership = true;
+    } ];
+    ensureDatabases = [
+      "matrix-synapse"
+    ];
+  };
+
+  services.redis.servers.matrix-synapse = {
+    enable = true;
+  };
+
+  environment.systemPackages = with pkgs; [
+    matrix-synapse-tools.synadm
+  ];
+
+  # create /run/matrix-synapse for the unix socket listener
+  systemd.services.matrix-synapse.ServiceConfing.RuntimeDirectory = "matrix-synapse";
+
+  services.matrix-synapse = {
+    enable = true;
+    withJemalloc = true;
+
+    # https://github.com/element-hq/synapse/blob/master/docs/usage/configuration/config_documentation.md
+    settings = {
+      server_name = "nixos.org";
+      public_base = "https://matrix.nixos.org";
+      admin_contact = "infra@nixos.org";
+
+      allow_public_rooms_over_federation = true;
+      allow_public_rooms_without_auth = true;
+
+      max_upload_size = "50M";
+
+      database = {
+        name = "psycopg2";
+        args = {
+          host = "/run/postgresql";
+        };
+      };
+
+      redis = {
+        enabled = true;
+        path = config.services.redis.matrix-synapse.path;
+      };
+
+      listeners = [ {
+        type = "http";
+        path = "/run/matrix-synapse/matrix-synapse.sock";
+        mode = "0660";
+        resources = [ {
+          compress = true;
+          names = [
+            "client"
+          ];
+        } {
+          compress = false;
+          names = [
+            "federation"
+          ];
+        } ];
+      } {
+        type = "http";
+        port = 8090;
```

Comment:
```suggestion
        port = 8090;
        bind_addresses = [ "127.0.0.1" ];
```

Reply:
> bind_addresses: a list of local addresses to listen on. The default is 'all local interfaces'.

https://github.com/element-hq/synapse/blob/master/docs/usage/configuration/config_documentation.md#listeners

Reply:
The default bind is on IPv4/IPv6 wildcard, so public.

https://github.com/element-hq/synapse/blob/v1.99.0/synapse/config/server.py#L57

I configured IPv4/IPv6 localhost instead.